web/src/components/Chat.tsx

```diff
@@ -72,28 +72,27 @@ const Chat = ({ messages, submitMessage, hasError, isLoading }: ChatProps): Reac
   }
 
   const submitOnEnter = (event: KeyboardEvent) => {
-    if (event.key === 'Enter' && !event.shiftKey) {
+    const isEnterAllowed = event.key === 'Enter' && !event.shiftKey && textInput.length > 0 && !isLoading && !hasError
+    if (isEnterAllowed) {
       event.preventDefault()
       onSubmit()
     }
   }
 
-  if (isLoading && !hasError) {
-    return (
-      <LoadingContainer>
-        <StyledLoadingSpinner />
-        <LoadingText>{t('loadingText')}</LoadingText>
-      </LoadingContainer>
-    )
-  }
-
   return (
     <Container>
-      <StyledChatConversation
-        $height={deviceHeight - chatInputContainerHeight}
-        messages={messages}
-        hasError={hasError}
-      />
+      {isLoading && !hasError ? (
+        <LoadingContainer>
+          <StyledLoadingSpinner />
+          <LoadingText>{t('loadingText')}</LoadingText>
+        </LoadingContainer>
+      ) : (
+        <StyledChatConversation
+          $height={deviceHeight - chatInputContainerHeight}
+          messages={messages}
+          hasError={hasError}
+        />
+      )}
```

Comment:
❓ Why did you change this? This leads to jumpy behavior as the chat input is already shown while loading and is only pushed down once it has finished loading.

Reply:
At least the current implementation makes the UI jump, so either revert it or try to fix it :)